src/process_wrap.cc

```diff
@@ -121,8 +123,10 @@ class ProcessWrap : public HandleWrap {
     // options.uid
     Local<Value> uid_v = js_options->Get(env->uid_string());
     if (uid_v->IsInt32()) {
-      int32_t uid = uid_v->Int32Value();
-      if (uid & ~((uv_uid_t) ~0)) {
+      int32_t uid_i32 = uid_v->Int32Value();
+      uv_uid_t uid = static_cast<uv_uid_t>(uid_i32);
+      // uv_uid_t may be unsigned, so compare with the widest available type
+      if (static_cast<intmax_t>(uid_i32) != static_cast<intmax_t>(uid)) {
```

Comment:
`uv_uid_t` is unsigned on most platforms (all supported platforms, actually) but e.g. on OS X, the entry for user `nobody` looks like this:
```
$ grep nobody /etc/passwd 
nobody:*:-2:-2:Unprivileged User:/var/empty:/usr/bin/false
```
Passing that works (or should work) because signed-to-unsigned conversion of -2 is `UINT_MAX-1` (and has well-defined behavior; the other way around is implementation-defined behavior.)

I remember we've had to rework similar uid/gid checks because they broke working code.